[automate-2530] document supported LDAP/SAML configurations (#2636)
chef/automate: dev-docs/auth.md

For further questions, the team can be reached at [#auth-team](https://chefio.slack.com/messages/C61F9HHKK/)

## Supported Identity Management Systems

> OKTA with SAML and AD with LDAP are the most common configurations among our users.

Reviewer: `AD with LDAP are the most common configurations among our users.` This is a bit misleading since most people using AD are using the AD specific configuration instead of LDAP. I understand what this is saying but the world `configurations` might be a bit overloaded here.

Reviewer: Oh just also realized this was a dev doc. That's an ignorable nit in that case.

> OKTA with SAML and AD with LDAP have anecdotally been reported to be the most common configurations among our users.

Reviewer 2: ```suggestion
OKTA with SAML and AD with LDAP are the most common configurations among our users.
```

We have also had success with the following configurations:

### SAML

- OKTA
- Ping
- OneLogin
- Office365
- Tivoli Federated Identity Manager

### LDAP

- Microsoft Active Directory (MSAD)

> ### Configurations with Known Issues

Reviewer 2: lets move this section to somewhere in the repo instead of in the docs

Author: these are dev docs so they're not part of the docs we deploy to customers. but i could move it to the authn README if that seems a little more out of the way

Reviewer 2: oh lol sorry, ignore me.

- Azure AD with SAML: we cannot support SAML with Azure AD because Automate cannot consume SAML IdP metadata, which Azure relies on to rotate keys.
- Appleconnect with SAML: invalid signature error.

> - in general, dex does not support idP initiated SSO

Reviewer 2: lets link to the issue in progress for this
https://github.com/dexidp/dex/pull/1514
